### PR TITLE
Bug/vote 1477 untranslated links

### PR DESCRIPTION
**Pull request by clmedders**
**_Delete all details that do not apply to this PR!_**

## Jira ticket (required)
[Vote-1477](https://bixal-projects.atlassian.net/jira/software/c/projects/VOTE/boards/254?modal=detail&selectedIssue=VOTE-1477&assignee=611272e0eb7aef00694b8a97)

## Description (optional)
This PR will add conditional logic for state pages for untranslated links.  The idea is that if a translation does not have a link in that language and is left blank in the CMS then the sting translation will concat (in English) for it so the user will know how ever when there is a link in the overide field that is translated it will remove the (in English) string

### Testing steps
1. Pull down branch
2. start local 
3. run `lando retune`
4. run `lando drush uli` and log in as a super user 
5. visit the site and navigate to  `Alabama` state page
6. click through each translation and you should see (in English) at the end of each link 
7. choose a few languages (Spanish and Talalog are the best examples) and click the edit button. 
8. add `http://example.com` to each override field
9. return back to the page and verify that the (in English) string is removed

>**Note**
> I was not able to check the other templates with updating the links I received this error message 
>  "Non-translatable fields can only be changed with updating the original language 
> I was able to check that the logic was working before updating the string translations and there were double instances of (in English) not sure why `Alamaba` let me know it but not others

### CODE REVIEW

File: web/themes/custom/vote_gov/templates/partial/register/registration-type--online.html.twig [hunk 1]
```diff
@@ -4,8 +4,16 @@
   <h2>{{ 'How to register to vote in @state_name' | t({ '@state_name': state_name }) }}</h2>
     {# registration link, more info link #}
     {% if content.field_registration_link | render %}
+    
       {% set registration_link = content.field_override_registration_link | field_value | render | default( content.field_registration_link | field_value | render ) %}
+
       {% set registration_link_text = "Start your online registration" | t %}
+
+      {% if language != "en" and content.field-override-registration-link is empty %}
```
Comment (rayestrada) [inline web/themes/custom/vote_gov/templates/partial/register/registration-type--online.html.twig:12]: Update to
`{% if language != "en" and not content.field_override_registration_link | render %}`

File: web/themes/custom/vote_gov/templates/partial/register/registration-type--online.html.twig [hunk 2]
```diff
@@ -4,8 +4,16 @@
...
+          {% set in_english = "(in English)" | t %}
+          {% set registration_link_text = registration_link_text ~ in_english %}
```
Comment (rayestrada) [inline web/themes/custom/vote_gov/templates/partial/register/registration-type--online.html.twig:14]: We need to add a space here 
`{% set registration_link_text = registration_link_text ~ ' ' ~ in_english %}`

**Review by rayestrada — CHANGES_REQUESTED** [review]

File: web/modules/custom/vote_utility/translations/nv.po
Comment (clmedders) [inline web/modules/custom/vote_utility/translations/nv.po:1]: I was not sure about this one but let me know if I didn't not do this one correctly

File: web/themes/custom/vote_gov/templates/partial/register/registration-type--mail.html.twig [hunk 3]
```diff
@@ -6,6 +6,12 @@
     {% if content.field_more_info_link | render %}
       {% set more_info_link = content.field_override_more_info_link | field_value | render | default( content.field_more_info_link | field_value | render ) %}
       {% set more_info_link_text = "Find out ways to register" | t %}
+
+    {% if language != "en" and not content.field_override_more_info_link  | render %}
+          {% set in_english = "(in English)" | t %}
```
Comment (rayestrada) [inline web/themes/custom/vote_gov/templates/partial/register/registration-type--mail.html.twig:11]: same here

File: web/themes/custom/vote_gov/templates/partial/register/registration-type--mail.html.twig [hunk 4]
```diff
@@ -6,6 +6,12 @@
...
+          {% set more_info_link = more_info_link ~ ' ' ~ in_english %}
```
Comment (rayestrada) [inline web/themes/custom/vote_gov/templates/partial/register/registration-type--mail.html.twig:12]: This should be `more_info_link_text`

File: web/themes/custom/vote_gov/templates/partial/register/registration-type--mail.html.twig [hunk 5]
```diff
@@ -24,6 +30,12 @@
 {% if content.field_confirm_registration_link | render %}
   {% set confirm_reg_link = content.field_override_confirm_reg_link | field_value | render | default( content.field_confirm_registration_link | field_value | render ) %}
   {% set confirm_reg_link_text = "confirm your voter registration status" | t %}
+
+    {% if language != "en" and not content.field_override_confirm_reg_link  | render %}
+          {% set in_english = "(in English)" | t %}
```
Comment (rayestrada) [inline web/themes/custom/vote_gov/templates/partial/register/registration-type--mail.html.twig:35]: same here

File: web/themes/custom/vote_gov/templates/partial/register/registration-type--online.html.twig [hunk 6]
```diff
@@ -4,8 +4,16 @@
   <h2>{{ 'How to register to vote in @state_name' | t({ '@state_name': state_name }) }}</h2>
     {# registration link, more info link #}
     {% if content.field_registration_link | render %}
+
```
Comment (rayestrada) [inline web/themes/custom/vote_gov/templates/partial/register/registration-type--online.html.twig:7]: remove extra line space

File: web/themes/custom/vote_gov/templates/partial/register/registration-type--online.html.twig [hunk 7]
```diff
@@ -4,8 +4,16 @@
   <h2>{{ 'How to register to vote in @state_name' | t({ '@state_name': state_name }) }}</h2>
     {# registration link, more info link #}
     {% if content.field_registration_link | render %}
+    
       {% set registration_link = content.field_override_registration_link | field_value | render | default( content.field_registration_link | field_value | render ) %}
+
```
Comment (rayestrada) [inline web/themes/custom/vote_gov/templates/partial/register/registration-type--online.html.twig:9]: remove extra line space

File: web/themes/custom/vote_gov/templates/partial/register/registration-type--online.html.twig [hunk 8]
```diff
@@ -4,8 +4,16 @@
...
       {% set registration_link_text = "Start your online registration" | t %}
+
+      {% if language != "en" and not content.field_override_registration_link | render %}
+          {% set in_english = "(in English)" | t %}
```
Comment (rayestrada) [inline web/themes/custom/vote_gov/templates/partial/register/registration-type--online.html.twig:13]: same here

File: web/themes/custom/vote_gov/templates/partial/register/registration-type--online.html.twig [hunk 9]
```diff
@@ -15,6 +23,12 @@
     {% if content.field_more_info_link | render %}
       {% set more_info_link = content.field_override_more_info_link | field_value | render | default( content.field_more_info_link | field_value | render ) %}
       {% set more_info_link_text = "register to vote by mail or in person" | t %}
+
+      {% if language != "en" and not content.field_override_more_info_link | render %}
+          {% set in_english = "(in English)" | t %}
```
Comment (rayestrada) [inline web/themes/custom/vote_gov/templates/partial/register/registration-type--online.html.twig:28]: same here

File: web/themes/custom/vote_gov/templates/partial/register/registration-type--online.html.twig [hunk 10]
```diff
@@ -34,6 +48,12 @@
 {% if content.field_confirm_registration_link | render %}
   {% set confirm_reg_link = content.field_override_confirm_reg_link | field_value | render | default( content.field_confirm_registration_link | field_value | render ) %}
   {% set confirm_reg_link_text = "confirm your voter registration status" | t %}
+
+    {% if language != "en" and not content.field_override_confirm_reg_link  | render %}
+          {% set in_english = "(in English)" | t %}
```
Comment (rayestrada) [inline web/themes/custom/vote_gov/templates/partial/register/registration-type--online.html.twig:53]: same here

File: testing/package-lock.json
Comment (rayestrada) [inline testing/package-lock.json:1]: Remove this file modification

**Review by rayestrada — CHANGES_REQUESTED** [review]

File: web/modules/custom/vote_utility/translations/es.po
[the remark this answers is not in the document]
Comment (clmedders) [reply web/modules/custom/vote_utility/translations/es.po:1]: removed (in english)

**Review by rayestrada — APPROVED** [review]

**Pull request merged**